### PR TITLE
11 reviews queries

### PR DESCRIPTION
**Pull request by MikeDobinson**
Starting to freak about how little time is left

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ const seed = require('../db/seeds/seed');
 const testData = require('../db/data/test-data/');
 const connection = require('../db/connection');
 const { expect } = require('@jest/globals');
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: is this a VSCode auto-add. Can probably safely be destroyed

File: __tests__/app.test.js [hunk 2]
```diff
@@ -39,28 +40,95 @@ describe('invalid endpoint', () => {
...
+          .then(({ body }) => {
+            const { msg } = body;
+            expect(msg).toBe('Invalid query');
+          });
+      });
+    });
+    describe('sorted_by', () => {
+      it('should change the key that the array is sorted by', () => {
+        return request(app)
+          .get('/api/reviews?sorted_by=review_id')
```
Comment (alexdevdavis) [inline __tests__/app.test.js]: I think the ticket asked for `sort_by`
At this scale, no big deal, but when you're developing software on a larger scale, naming according to the ticket helps avoid a heap of strife.

File: models/reviews.models.js [hunk 3]
```diff
@@ -18,25 +18,67 @@ exports.fetchReviewById = (reviewId) => {
...
+    fetchAllReviewsSQL += ` 
+    WHERE reviews.category = $1 `;
+    queryParameters.push(category);
+  }
+
+  fetchAllReviewsSQL += `GROUP BY reviews.review_id 
+  ORDER BY ${sorted_by} ${order}`;
+
+  return db.query(fetchAllReviewsSQL, queryParameters).then(({ rows }) => {
+    if (rows.length === 0) {
```
Comment (alexdevdavis) [inline models/reviews.models.js]: `!rows.length` achieves the same here

File: models/reviews.models.js [hunk 4]
```diff
@@ -18,25 +18,67 @@ exports.fetchReviewById = (reviewId) => {
...
+    }
+  });
 };
 
 exports.fetchCommentsByReviewId = (reviewId) => {
   return db
     .query(
       `SELECT * FROM comments 
-       WHERE review_id = $1 
+       WHERE review_id = ($1)
```
Comment (alexdevdavis) [inline models/reviews.models.js]: don't think `$1` needs the round brackets

**Review by alexdevdavis — APPROVED** [review]
Nicely done. PR approved 😎 
- would be good to see a test that checks the three queries all work together, from one URL string
- see a few refactor suggestions

**Comment by MikeDobinson** [comment]
Fixed all these issues, thanks

**Pull request merged**